src/controller/src/clusters.rs

```diff
@@ -83,12 +83,27 @@ pub struct ReplicaAllocation {
     /// The number of credits per hour that the replica consumes.
     #[serde(deserialize_with = "mz_repr::adt::numeric::str_serde::deserialize")]
     pub credits_per_hour: Numeric,
+    /// Whether instances of this type can be created
+    #[serde(default = "return_true")]
+    pub enabled: bool,
+}
+
+/// This method exists solely for the purpose of making ReplicaAllocation enabled by
+/// default. The default value of a bool is false, and the only
+/// way to make an object take on any other value by default is to pass it a
+/// function that returns the desired default value. See
+/// <https://github.com/serde-rs/serde/issues/1030>
+#[inline(always)]
+fn return_true() -> bool {
+    true
```

Comment:
It's up to you, but you can avoid this hack if you want by renaming `enabled` to `disabled` and inverting the logic elsewhere in the PR.

Reply:
Done, thanks for the suggestion!